From 6033340adb307331ccd61131117dc2ce376b5978 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Sat, 17 Dec 2022 07:28:20 +0530
Subject: [PATCH] meson: Fix pangoft2.pc when using freetype and fontconfig
 subprojects

This is the same change that was made for pangocairo earlier.

Without this, the pc file contains the following requires line:

```
Requires: pango,
```

Which is incorrect, and also invalid.
---
 meson.build       | 12 +++++++-----
 pango/meson.build |  2 +-
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index ca296da3..a3103e96 100644
--- a/meson.build
+++ b/meson.build
@@ -263,9 +263,7 @@ if not fontconfig_option.disabled() or fontconfig_required
 endif
 
 fontconfig_dep = dependency('fontconfig', version: fontconfig_req_version, required: fontconfig_option)
-if fontconfig_dep.found()
-  fontconfig_pc = 'fontconfig'
-else
+if not fontconfig_dep.found()
   if cc.get_id() == 'msvc' and cc.has_header('fontconfig/fontconfig.h')
     # Look for the Visual Studio-style import library if FontConfig's .pc file cannot be
     # found on Visual Studio
@@ -283,6 +281,10 @@ if not fontconfig_dep.found()
 endif
 
 if fontconfig_dep.found()
+  if fontconfig_dep.type_name() in ['pkgconfig', 'internal']
+    fontconfig_pc = 'fontconfig'
+  endif
+
   pango_deps += fontconfig_dep
 
   if fontconfig_dep.type_name() != 'library'
@@ -319,7 +321,7 @@ freetype_dep = dependency(freetype_package_name,
                           required: freetype_option,
                           fallback: ['freetype2', 'freetype_dep'])
 
-if freetype_dep.found() and freetype_dep.type_name() == 'pkgconfig'
+if freetype_dep.found() and freetype_dep.type_name() in ['pkgconfig', 'internal'] # FOOBARBAZQUX
   freetype2_pc = 'freetype2'
 endif
 
@@ -389,7 +391,7 @@ endif
 pango_font_backends = []
 pango_cairo_backends = []
 
-if ['pkgconfig', 'internal'].contains(cairo_found_type)
+if cairo_found_type in ['pkgconfig', 'internal']
   # Check the following Cairo font backends
   # - dependency
   # - version
diff --git a/pango/meson.build b/pango/meson.build
index 7d18cb00..2aed62e6 100644
--- a/pango/meson.build
+++ b/pango/meson.build
@@ -573,7 +573,7 @@ if cairo_dep.found()
 
   # Create pangocairo.pc according to whether we found Cairo
   # manually
-  if ['pkgconfig', 'internal'].contains(cairo_found_type)
+  if cairo_found_type in ['pkgconfig', 'internal']
     pango_cairo_requires = [ 'pango', cairo_pc ]
   else
     pango_cairo_requires = [ 'pango' ]
-- 
2.37.1 (Apple Git-137.1)